src/tests/roc_fec/test_writer_reader.cpp

```diff
@@ -2984,27 +3095,177 @@ TEST(writer_reader, reader_invalid_fec_scheme_repair_packet) {
 
         // drop other repair packets
         for (size_t i = 0; i < NumRepairPackets - NumRepairPackets / 2 - 1; ++i) {
-            packet::PacketPtr p = writer_queue.read();
+            packet::PacketPtr p;
+            UNSIGNED_LONGS_EQUAL(status::StatusOK, writer_queue.read(p));
             CHECK(p);
             CHECK((p->flags() & packet::Packet::FlagRepair) != 0);
         }
 
         // deliver more source packets
         for (size_t i = 0; i < NumSourcePackets; ++i) {
-            packet::PacketPtr p = writer_queue.read();
+            packet::PacketPtr p;
+            UNSIGNED_LONGS_EQUAL(status::StatusOK, writer_queue.read(p));
             CHECK(p);
             CHECK((p->flags() & packet::Packet::FlagRepair) == 0);
             source_queue.write(p);
         }
         UNSIGNED_LONGS_EQUAL(NumSourcePackets, source_queue.size());
 
         // reader should shut down
-        CHECK(!reader.read());
+        packet::PacketPtr pp;
+        UNSIGNED_LONGS_EQUAL(status::StatusNoData, reader.read(pp));
+        CHECK(!pp);
         CHECK(!reader.is_alive());
         UNSIGNED_LONGS_EQUAL(0, source_queue.size());
         UNSIGNED_LONGS_EQUAL(0, repair_queue.size());
     }
 }
 
+TEST(writer_reader, failed_to_read_source_packet) {
+    for (size_t n_scheme = 0; n_scheme < CodecMap::instance().num_schemes(); ++n_scheme) {
+        codec_config.scheme = CodecMap::instance().nth_scheme(n_scheme);
+
+        const status::StatusCode codes[] = {
+            status::StatusUnknown,
+        };
+
+        for (size_t n_code = 0; n_code < ROC_ARRAY_SIZE(codes); ++n_code) {
```

Comment:
Do you think we need a loop over codes in these tests?

It seems that all codes except NoData are handled the same way, so it should be enough to test NoData + any single code. NoData is covered by other tests (because Queue returns it), and I guess these tests can check just one another code.

Reply:
(same for 2 tests below)